### PR TITLE
bake: add network mode support to HCL and docs for network and entitlements

### PR DESCRIPTION
**Pull request by tonistiigi**
follow-up #2666

### CODE REVIEW

File: docs/bake-reference.md [hunk 1]
```diff
@@ -749,6 +768,27 @@ target "app" {
 }
 ```
 
+### `target.network`
+
+Specify the network mode for the whole build request. This will override the the default network mode
```
Comment (crazy-max) [inline docs/bake-reference.md:773]:
```suggestion
Specify the network mode for the whole build request. This will override the default network mode
```

**Review by crazy-max — CHANGES_REQUESTED** [review]
Small typo otherwise LGTM :rocket:

@dvdksn PTAL for copy

File: docs/bake-reference.md [hunk 2]
```diff
@@ -505,6 +505,25 @@ $ docker buildx bake --print -f - <<< 'target "default" {}'
 }
 ```
 
+### `target.entitlements`
+
+Entitlements are permissions that the build process requires to run.
+
+Currently supported entitlements are:
+
+- `network.host`: Allows the build to use commands that access the host network. In Dockerfile, use `RUN --network=host` to run a command with host network enabled.
```
Comment (dvdksn) [inline docs/bake-reference.md:514]:
```suggestion
- `network.host`: Allows the build to use commands that access the host network. In Dockerfile, use [`RUN --network=host`](https://docs.docker.com/reference/dockerfile/#run---networkhost) to run a command with host network enabled.
```

File: docs/bake-reference.md [hunk 3]
```diff
@@ -505,6 +505,25 @@ $ docker buildx bake --print -f - <<< 'target "default" {}'
...
+
+- `security.insecure`: Allows the build to run commands in privileged containers that are not limited by the default security sandbox. Such container may potentially access and modify system resources. In Dockerfile, use `RUN --security=insecure` to run a command in a privileged container.
```
Comment (dvdksn) [inline docs/bake-reference.md:516]:
```suggestion
- `security.insecure`: Allows the build to run commands in privileged containers that are not limited by the default security sandbox. Such container may potentially access and modify system resources. In Dockerfile, use [`RUN --security=insecure`](https://docs.docker.com/reference/dockerfile/#run---security) to run a command in a privileged container.
```

**Review by dvdksn — APPROVED** [review]

**Review by crazy-max — APPROVED** [review]

**Pull request merged**